text/0000-target-tier-policy.md

```diff
@@ -0,0 +1,685 @@
+    rust-lang/rust repository), or whether the dependency is a native library
+    or binary. In other words, the introduction of the target must not cause a
+    user installing or running a version of Rust or the Rust tools to be
+    subject to any new license requirements.
+  - If the target supports building host tools (such as `rustc` or `cargo`),
+    those host tools must not depend on proprietary (non-FOSS) libraries, other
+    than ordinary runtime libraries supplied by the platform and commonly used
+    by other binaries built for the target. For instance, `rustc` built for the
+    target may depend on a common proprietary C runtime library or console
+    output library, but must not depend on a proprietary code generation
+    library or code optimization library. Rust's license permits such
+    combinations, but the Rust project has no interest in maintaining such
+    combinations within the scope of Rust itself, even at tier 3.
+  - New targets should not require proprietary (non-FOSS) components to link a
+    functional binary or library.
+  - "onerous" here is an intentionally subjective term. At a minimum, "onerous"
+    legal/licensing terms include but are *not* limited to: non-disclosure
+    requirements, non-compete requirements, contributor license agreements
+    (CLAs) or equivalent, "non-commercial"/"research-only"/etc terms,
+    requirements conditional on the employer or employment of any particular
+    Rust developers, revocable terms, any requirements that create liability
+    for the Rust project or its developers or users, or any requirements that
+    adversely affect the livelihood or prospects of the Rust project or its
+    developers or users.
+- Neither this policy nor any decisions made regarding targets shall create any
+  binding agreement or estoppel by any party. If any member of an approving
+  Rust team serves as one of the maintainers of a target, or has any legal or
+  employment requirement (explicit or implicit) that might affect their
+  decisions regarding a target, they must recuse themselves from any approval
+  process regarding the target's tier status, though they may otherwise
+  participate in discussions.
+  - This requirement does not prevent part or all of this policy from being
+    cited in an explicit contract or work agreement (e.g. to implement or
+    maintain support for a target). This requirement exists to ensure that a
+    developer or team responsible for reviewing and approving a target does not
+    face any legal threats or obligations that would prevent them from freely
+    exercising their judgment in such approval, even if such judgment involves
+    subjective matters or goes beyond the letter of these requirements.
+- Tier 3 targets should attempt to implement as much of the standard libraries as
+  possible and appropriate (`core` for most targets, `alloc` for any target
+  with a standard memory allocator, `std` for targets with an operating
```

Comment:
AFAIK, `alloc` is available as soon as `core` is, and you're supposed to plug in your own memory allocator, not use a standard one.

Reply:
I meant "standard" as "target-supplied" here; I didn't realize this might be ambiguous.

In any case, I don't think the target tier policy should be prescriptive about *how* to implement `alloc` for a target. I've reworded this accordingly.